server/src/main/java/org/elasticsearch/index/IndexVersion.java

```diff
@@ -117,15 +117,16 @@ private static IndexVersion registerIndexVersion(int id, Version luceneVersion,
     public static final IndexVersion V_8_9_0 = registerIndexVersion(8_09_00_99, Version.LUCENE_9_7_0, "32f6dbab-cc24-4f5b-87b5-015a848480d9");
     public static final IndexVersion V_8_9_1 = registerIndexVersion(8_09_01_99, Version.LUCENE_9_7_0, "955a80ac-f70c-40a5-9399-1d8a1e5d342d");
     public static final IndexVersion V_8_10_0 = registerIndexVersion(8_10_00_99, Version.LUCENE_9_7_0, "2e107286-12ad-4c51-9a6f-f8943663b6e7");
-    public static final IndexVersion V_8_11_0 = registerIndexVersion(8_11_00_99, Version.LUCENE_9_7_0, "f08382c0-06ab-41f4-a56a-cf5397275627");
+    public static final IndexVersion V_8_11_0 = registerIndexVersion(8_11_00_99, Version.LUCENE_9_8_0, "f08382c0-06ab-41f4-a56a-cf5397275627");
 
     /*
      * READ THE COMMENT BELOW THIS BLOCK OF DECLARATIONS BEFORE ADDING NEW INDEX VERSIONS
      * Detached index versions added below here.
      */
-    public static final IndexVersion V_8_500_000 = registerIndexVersion(8_500_000, Version.LUCENE_9_7_0, "bf656f5e-5808-4eee-bf8a-e2bf6736ff55");
-    public static final IndexVersion V_8_500_001 = registerIndexVersion(8_500_001, Version.LUCENE_9_7_0, "45045a5a-fc57-4462-89f6-6bc04cda6015");
-    public static final IndexVersion V_8_500_002 = registerIndexVersion(8_500_002, Version.LUCENE_9_7_0, "50b39bf8-6c6a-443e-a5e5-069438d843c1");
+    public static final IndexVersion V_8_500_000 = registerIndexVersion(8_500_000, Version.LUCENE_9_8_0, "bf656f5e-5808-4eee-bf8a-e2bf6736ff55");
+    public static final IndexVersion V_8_500_001 = registerIndexVersion(8_500_001, Version.LUCENE_9_8_0, "45045a5a-fc57-4462-89f6-6bc04cda6015");
+    public static final IndexVersion V_8_500_002 = registerIndexVersion(8_500_002, Version.LUCENE_9_8_0, "50b39bf8-6c6a-443e-a5e5-069438d843c1");
```

Comment:
This looks odd to me, perhaps we need to revise it because we can't change existing index versions?

Reply:
🤔 its almost like there should be a `V_8_6....` for the Lucene bump. Especially if these versions match what we are doing for transport:

```
     * M_NNN_SS_P
     *
     * M - The major version of Elasticsearch
     * NNN - The server version part
     * SS - The serverless version part. It should always be 00 here, it is used by serverless only.
     * P - The patch version part
```

The inconsistency between the two confuses me.

Reply:
Yes, the IndexVersion does need to move to a similar scheme as TransportVersion - @rjernst is this something you're currently working on?

Reply:
Yes, we'll need to adjust IndexVersion as we did TransportVersion. It's not something I'm currently working on, but I'm happy to pick it up.

Reply:
I added a 003 version for now that only brings in the Lucene upgrade.